### PR TITLE
Add the complete Flame linter and fixes issues

### PR DESCRIPTION
**Pull request by luanpotter**

### CODE REVIEW

**Comment by luanpotter** [comment]
(I will switch the required GH action once approved)

**Review by erickzanardo — APPROVED** [review]
I am unsure if we need the same bash scripts from flame here, they are more complex because they were made to work with multiples projects on the same repo, here, we have only a single repository.

Either way I don't have strong opinions, so is up to you.

File: test/tmx_object_test.dart [hunk 1]
```diff
@@ -117,16 +117,18 @@ void main() {
         expect(ps[2].y, equals(Point(x: -78, y: 19).y));
       });
 
-      test('sets isPolygon to true',
-          () => expect(tiledObject.isPolygon, isTrue));
+      test(
+        'sets isPolygon to true',
+        () => expect(tiledObject.isPolygon, isTrue),
+      );
```
Comment (luanpotter) [inline test/tmx_object_test.dart:123]: @erickzanardo even my tired eyes are not good enough, you can see I manually fixed every file on my previous PR and stuff was left unformatted...

File: analysis_options.yaml [hunk 2]
```diff
@@ -1,68 +1,147 @@
 # Source of linter options:
 # http://dart-lang.github.io/linter/lints/options/options.html
 
+analyzer:
+  strong-mode:
+    implicit-casts: false
+    implicit-dynamic: false
+
+  plugins:
+    - dart_code_metrics
```
[the remark this answers is not in the document]
Comment (luanpotter) [reply analysis_options.yaml:10]: https://github.com/flame-engine/tiled.dart/pull/22/files#r638223887

**Comment by luanpotter** [comment]
We always have at least the example as well right? so that is already 2. and if we ever need to add more, it is out of the box. the main reason though was because I didn't want to rewrite anything, I think it's good being consistent so we can make sure we don't forget any step on any repo (like dartdoc, etc).

File: analysis_options.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (erickzanardo) [reply analysis_options.yaml:10]: I rather have some few lines without trailing commas, than having my editor going out of whack because of the plugin :(

**Comment by erickzanardo** [comment]
> We always have at least the example as well right? so that is already 2. and if we ever need to add more, it is out of the box. the main reason though was because I didn't want to rewrite anything, I think it's good being consistent so we can make sure we don't forget any step on any repo (like dartdoc, etc).

Yeah, but the linter/formatter automatically reads the example folder 😉 , so for that case, nothing special is needed

**Comment by luanpotter** [comment]
good to know! I will leave as is for now, if the complexity cause trouble, I will write simple versions.

eventually we could find a way to make generic scripts across repos

**Pull request merged**